airbyte-lib/airbyte_lib/types.py

```diff
@@ -41,7 +41,10 @@ def _get_airbyte_type(json_schema_property_def: dict[str, str | dict]) -> tuple[
     json_schema_type = json_schema_property_def.get("type", None)
 
     if json_schema_type in ["string", "number", "boolean", "integer"]:
-        return cast(str, json_schema_type), None
+        return json_schema_type, None
+
+    if json_schema_type == "object" and "properties" in json_schema_property_def:
```

Comment:
1. The removal of "cast()" was not needed, but my linter was correctly telling me that the "if" condition already established the type, so the cast was redundant in that context.
2. The main update here is checking for "object" type, which I had not handled in my initial implementation. I added it because the "faker" dataset does use nested objects to represent addresses in the users table, and without this code change, the exception below ("Could not determine airbyte type from JSON schema type: object") was reached.